alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs

```diff
@@ -330,18 +342,68 @@ hashSD (w@(TxWitnessConstr (Memo (TxWitnessRaw _ _ scriptdata) _))) =
     else Just (ScriptDataHash (hashAnnotated scriptdata))
 -}
 
+-- ========================================================================
+-- What is called ScriptDataHash is called WitnessLangDataHash
+-- This is a hash of two things. We make a MemoBytes of these two things
+-- so that we can hash it.
+
+data WitnessLangDataRaw era = WitnessLangDataRaw (Map.Map RdmrPtr (Data era)) (Set (PPHash (Crypto era)))
+  deriving (Show, Eq, Generic, Typeable)
+
+deriving instance NoThunks (WitnessLangDataRaw era)
+
+instance Era era => ToCBOR (WitnessLangDataRaw era) where
+  toCBOR (WitnessLangDataRaw m s) = encode (Rec WitnessLangDataRaw !> To m !> To s)
+
+instance Era era => FromCBOR (Annotator (WitnessLangDataRaw era)) where
+  fromCBOR =
+    decode
+      ( Ann (RecD WitnessLangDataRaw)
+          <*! D (decodeMapTraverse (pure <$> fromCBOR) fromCBOR)
+          <*! Ann From
+      )
+
+newtype WitnessLangData era = WitnessLangDataConstr (MemoBytes (WitnessLangDataRaw era))
```

Comment:
OK, I think I can see what this is, but can you add some documentation explaining this in the context of what it's used for?

Reply:
This has been renamed to WitnessPPData, and is now explainded as follows:
`-- A WitnessPPDataHash is the hash of two things. The first part comes from
-- the witnesses and the second comes from the Protocol Parameters (PParams).
-- In order to hash 2 things we make a newtype WitnessPPData which will be
-- a MemoBytes of these two things (WitnessPPDataRaw), so that we can hash it.`